ntp-proto/src/time_types.rs

```diff
@@ -141,9 +141,16 @@ impl NtpDuration {
     }
 
     pub(crate) const fn to_bits_short(self) -> [u8; 4] {
+        // serializing negative durations should never happen
+        // and indicates a programming error elsewhere.
+        // as for duration that are too large, saturating is
+        // the safe option.
         assert!(self.duration >= 0);
-        assert!(self.duration <= 0x0000FFFFFFFFFFFF);
-        (((self.duration & 0x0000FFFFFFFF0000) >> 16) as u32).to_be_bytes()
+        if self.duration > 0x0000FFFFFFFFFFFF {
```

Comment:
Style suggestion: I like to write these as a match, and you could extract the common `to_be_bytes()`:

```rust
        match self.duration > 0x0000FFFFFFFFFFFF {
            true => 0xFFFFFFFF as u32,
            false => ((self.duration & 0x0000FFFFFFFF0000) >> 16) as u32,
        }.to_be_bytes()
```

The match more neatly aligns the two outcomes which I usually find easier to read.

Reply:
good idea, implemented.